data/ui/book_detail.blp

```diff
@@ -111,74 +115,28 @@ template $BookDetail: Adw.NavigationPage {
                     ]
                   }
 
+                  ProgressBar book_progress_bar {
+                  }
+
                   Box {
-                    valign: end;
-                    margin-top: 18;
-                    spacing: 6;
+                    orientation: horizontal;
+                    hexpand: true;
+                    halign: fill;
 
-                    $ProgressRing book_progress_ring {
-                      valign: center;
+                    Label progress_label {
+                      halign: start;
                     }
 
-                    Grid {
+                    Label total_label{
+                      halign: end;
                       hexpand: true;
-                      row-spacing: 3;
-                      column-spacing: 20;
-
-                      Label {
-                        halign: start;
-                        hexpand: true;
-                        label: _("Remaining");
-
-                        styles [
-                          "dim-label",
-                        ]
-
-                        layout {
-                          column: 0;
-                          row: 0;
-                        }
-                      }
-
-                      Label remaining_label {
-                        hexpand: true;
-                        xalign: 0;
-
-                        layout {
-                          column: 1;
-                          row: 0;
-                        }
-                      }
-
-                      Label {
-                        halign: start;
-                        hexpand: true;
-                        label: _("Total");
-
-                        styles [
-                          "dim-label",
-                        ]
-
-                        layout {
-                          column: 0;
-                          row: 1;
-                        }
-                      }
-
-                      Label total_label{
-                        hexpand: true;
-                        xalign: 0;
-
-                        layout {
-                          column: 1;
-                          row: 1;
-                        }
-                      }
+                      label: "loading...";
```

Comment:
```suggestion
                      label: _("Calculating…");
```
`_()` is needed to make this string translatable. Also, we use an actual ellipsis (`…`) character, instead of the three dots (`...`)